opendcs-rest-api/src/main/java/org/opendcs/odcsapi/res/DataSourceResources.java

```diff
@@ -52,80 +59,285 @@ public class DataSourceResources
 	@RolesAllowed({AuthorizationCheck.ODCS_API_GUEST})
 	public Response getDataSourceRefs() throws DbException
 	{
-		Logger.getLogger(ApiConstants.loggerName).fine("getDataSourceRefs");
-		try (DbInterface dbi = new DbInterface();
-			ApiDataSourceDAO dao = new ApiDataSourceDAO(dbi))
+		try
+		{
+			dbIo = getLegacyDatabase();
+			DataSourceList dsl = new DataSourceList();
+			dbIo.readDataSourceList(dsl);
+			return Response.status(HttpServletResponse.SC_OK).entity(map(dsl)).build();
+		}
+		catch (DatabaseException ex)
+		{
+			throw new DbException("Error reading data source list: " + ex);
+		}
+		finally
+		{
+			dbIo.close();
+		}
+	}
+
+	static ArrayList<ApiDataSourceRef> map(DataSourceList dsl)
+	{
+		ArrayList<ApiDataSourceRef> ret = new ArrayList<>();
+		for(DataSource ds : dsl.getList())
 		{
-			return ApiHttpUtil.createResponse(dao.readDataSourceRefs());
+			ApiDataSourceRef adr = new ApiDataSourceRef();
+			if (ds.getId() != null)
+			{
+				adr.setDataSourceId(ds.getId().getValue());
+			}
+			else
+			{
+				adr.setDataSourceId(DbKey.NullKey.getValue());
+			}
+			adr.setName(ds.getName());
+			adr.setType(ds.dataSourceType);
+			adr.setUsedBy(ds.numUsedBy);
+			if (ds.getArguments() != null)
+			{
+				adr.setArguments(propsToString(ds.getArguments()));
+			}
+			else
+			{
+				adr.setArguments(ds.getDataSourceArg());
+			}
+			ret.add(adr);
 		}
+		return ret;
+	}
+
+	static String propsToString(Properties props)
+	{
+		if (props == null || props.isEmpty())
+			return null;
```

Comment:
Suggest braces

Reply:
FYI, the braces are required. you'll see them in old code, but definitely don't add them to new, and if you're tweaking like the entire method or class of old code they should be updated.